context.go

```diff
@@ -21,25 +21,60 @@
 package fx
 
 import (
-	gcontext "context"
+	"context"
 
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-client-go"
+	"go.uber.org/fx/service"
 	"go.uber.org/fx/ulog"
 )
 
-// Context embeds Host and Go stdlib context for use
-type Context interface {
-	gcontext.Context
+type contextKey int
 
-	Logger() ulog.Log
+const _contextHost contextKey = iota
+
+type ctxHost struct {
+	log ulog.Log
+}
+
+func contextHost(ctx context.Context) ctxHost {
+	c := ctx.Value(_contextHost)
+	if c == nil {
+		c = ctxHost{}
+		ctx = context.WithValue(ctx, _contextHost, c)
+	}
+	return c.(ctxHost)
 }
 
-// NopContext is defined for testing purpose
-var NopContext = nopContext{gcontext.Background()}
+// SetContextHost sets the context with context aware logger
+func SetContextHost(ctx context.Context, host service.Host) context.Context {
```

Comment:
You could name this `NewContext`.

https://golang.org/pkg/context/#Context

It's slightly different than the example, because you don't expose the struct you're storing in the context, but I thought I'd point out the standard name.